Switch to JDownloader.jar download and update architecture support (Fixes #51)

EmKaCe opened the pull request:
Closes #51.

As reported in the issue, the JDownloader team has disabled the outdated `JD2Setup_x64.sh` installer script. They have provided a new static URL to fetch the `JDownloader.jar` file directly, which is all this flatpak needs.

This PR implements their requested changes:

- Removes the old `JD2Setup_x64.sh` source and the build steps required to run it.
- Adds the new static URL for `JDownloader.jar` as a primary source with the correct file size and sha256 hash
- Updates architecture support: Since we are now using the architecture-independent `.jar` file instead of an `x86_64`-specific shell script, this commit also updates the supported architectures (e.g., adding `aarch64`).

This should fix the build failures and simplifies the overall manifest.

EmKaCe committed:
Update architecture support and replace JD2Setup.sh with JDownloader.jar. Fixes #51

KuSh commented on flathub.json:
```diff
@@ -1,3 +1,3 @@
 {
-  "only-arches": ["x86_64"]
+  "only-arches": ["x86_64", "i386", "aarch64", "riscv64"]
```
You just can remove that file entirely

jiaz83 replied:
We either need to remove i386 because of build failing due to "Error: Unsupported architecture: i386" or remove that file completely

EmKaCe replied:
I would just go ahead and remove the file entirely

KuSh commented on jd-wrapper.sh:
```diff
@@ -1,13 +1,10 @@
-    mv ${JDDIR}/tmp/JDownloader.jar ${JDDIR}
-    rm -rf ${JDSETUP} ${JDDIR}/tmp
+    mkdir -p ${JDDIR}
+    cp /app/extra/JDownloader.jar ${JDDIR}/
```
No, files should directly be installed from the manifest file. Same for env variables if still needed. So this file should be removed too

jiaz83 replied:
Can you please add the additional changes? if ${JDDIR}/JDownloader.jar does not exist or is broken (can be tested with jar -tvf JDownloader.jar), 
then replace it with /app/extra/JDownloader.jar but also delete ${JDDIR}/Core.jar and ${JDDIR}/update folder and ${JDDIR}/tmp folder. 
Background: JDownloader.jar can be missing/broken when the update is interrupted in very bad timing/moment, leaving an inconsitent installation. when no Core.jar file exists, JDownloader will go into "fix installation" mode, see https://support.jdownloader.org/de/knowledgebase/article/fix-jdownloader-installation so the result will be a consistent installation again.

Thanks for your time and work!

EmKaCe replied:
From what I can see JDownloader.jar tries to write data in its install location, so having the JDownloader.jar file in the read-only directory breaks the package.

So from what I can tell we do need to copy to /app/extra/ and also as @jiaz83 mentioned will need to add the `${JDDIR}` lines to ensure updates don't break.

I hope I did get this correctly, since this is my first time working on a public flatpak package.
Please advise how we should approach this change, I am more than happy to implement those changes.

jiaz83 replied:
Having JDownloader.jar file in sources ensures correct/verified file/size/hash being used to install JDownloader. The approach to have JDownloader.jar in /app/extra directory and then copy over to ${JDDIR} looks good to me. Of course we could remove it from source completely and use wget/curl to download JDownloader.jar directly but then verification would need to be implemented in script manually -> increases complexity of script at no advantages.

KuSh replied:
No no I just thought it wasn't a self replacing jar. If it needs those extra steps, those should be implemented.

jiaz83 replied:
Thanks for clarification :)

EmKaCe replied:
> Can you please add the additional changes? if ${JDDIR}/JDownloader.jar does not exist or is broken (can be tested with jar -tvf JDownloader.jar), then replace it with /app/extra/JDownloader.jar but also delete ${JDDIR}/Core.jar and ${JDDIR}/update folder and ${JDDIR}/tmp folder. Background: JDownloader.jar can be missing/broken when the update is interrupted in very bad timing/moment, leaving an inconsitent installation. when no Core.jar file exists, JDownloader will go into "fix installation" mode, see https://support.jdownloader.org/de/knowledgebase/article/fix-jdownloader-installation so the result will be a consistent installation again.
> 
> Thanks for your time and work!

So do you mean like this?

```sh
#!/bin/bash

JDDIR=${XDG_DATA_HOME}/jdownloader

if [ ! -f ${JDDIR}/JDownloader.jar ] || ! jar -tvf ${JDDIR}/JDownloader.jar > /dev/null 2>&1; then
    rm -f ${JDDIR}/Core.jar
    rm -rf ${JDDIR}/update ${JDDIR}/tmp
    mkdir -p ${JDDIR}
    cp /app/extra/JDownloader.jar ${JDDIR}/
fi

exec java -jar ${JDDIR}/JDownloader.jar "${@}"
```

I replaced the JDownloader.jar with a broken file while testing to ensure that the behavior works as you described and from what I can see it does replace the JDownloader.jar and the Core.jar as intended.

jiaz83 replied:
@EmKaCe Thanks. Yes, that's what I had in mind. Most likely you can remove the "file exists" check as well as jar verification will fail for "file not exists" ;) Thanks for your time and work!

EmKaCe committed:
remove: flathub.json

EmKaCe committed:
fix: enhance JDownloader.jar check and cleanup in jd-wrapper.sh

EmKaCe committed:
Removed redundant file check

EmKaCe commented:
I have made some changes as discussed in the requested review:
- Removed `flathub.json`, should also fix #43 
- Added a check for validity of `JDownloader.jar`
- Re-added cleanup in case an update gets corrupted.

Feel free to let me know if anything else needs to be done, otherwise I would suggest merging the PR @KuSh @jiaz83

KuSh commented on jd-wrapper.sh:
```diff
@@ -1,13 +1,12 @@
-    rm -rf ${JDSETUP} ${JDDIR}/tmp
+if ! jar -tvf ${JDDIR}/JDownloader.jar > /dev/null 2>&1; then
+    rm -f ${JDDIR}/Core.jar
+    rm -rf ${JDDIR}/update ${JDDIR}/tmp
```
```suggestion
    rm -rf ${JDDIR}/JDownloader.jar ${JDDIR}/Core.jar ${JDDIR}/update ${JDDIR}/tmp 2>/dev/null
```

KuSh commented on jd-wrapper.sh:
```diff
@@ -1,13 +1,12 @@
+    rm -f ${JDDIR}/Core.jar
+    rm -rf ${JDDIR}/update ${JDDIR}/tmp
+    mkdir -p ${JDDIR}
+    cp /app/extra/JDownloader.jar ${JDDIR}/
```
```suggestion
    install -Dm644 /app/extra/JDownloader.jar ${JDDIR}
```

KuSh commented:
Thanks @EmKaCe I've made some further suggestions!

EmKaCe committed:
Safer deletion of outdated/broken files

Co-authored-by: Nicolas Le Cam <niko.lecam@gmail.com>

EmKaCe committed:
Use install instead of copy

Co-authored-by: Nicolas Le Cam <niko.lecam@gmail.com>
